modules/vault/options.go

```diff
@@ -0,0 +1,82 @@
+package vault
+
+import "fmt"
+
+// Option is a function type for modifying a Vault configuration
+type Option func(*Config)
+
+// Config is a struct that contains various configuration options for the Vault
+type Config struct {
+	imageName    string              // The name of the Docker image to use for the Vault
+	token        string              // The root token to use for the Vault
+	port         int                 // The port number to use for the Vault
```

Comment:
port is 8200 by default in Vault image. I don't think it is useful to change the port. Do you have any use case in mind? In Java, it is already deprecated

Reply:
I tried to add all functionality Java has. I didn't notice it is deprecated, I will remove it 👍